test/serializer/additional-functions/return-or-many-throw.js

```diff
@@ -31,7 +31,9 @@ function func1(doNotThrow) {
 }
 
 if (global.__optimize)
-  __optimize(func1);
+  __optimize(func1, {
+    pure: false,
```

Comment:
Why do we need this optional `pure` disabling feature? Would the tests not work with your changes? Is it just for testing?
I don't really want to maintain two optimized function modes.

Reply:
They won’t work with my changes as the function isn’t behaving pure and thus doesn’t go down the same code path to emit binding changes to objects that were mutated outside the scope.

Reply:
Of course, I see... Oh well. Leave behind a comment?